### PR TITLE
Fix #16453: Tile inspector invisibility shortcut does not use a game action

### LINKED ISSUE #16453: Tile inspector visibility hotkey does not use a game action
### Operating System

Windows 10, 64-bit

### OpenRCT2 build

OpenRCT2, v0.3.5.1 (730364d on develop) provided by GitHub

Still occurring in 0.4.4.

### Describe the issue

When using a hotkey to make an object visible or invisible in the tile inspector, it does not use a game action. This means that in multiplayer, when connected as a client, the host and other players will not see the invisibility toggled. It also will not persist in replays.

### Area(s) with issue?

New Save Format (NSF)

### Steps to reproduce

1. Set a hotkey for "Tile inspector: Toggle invisibility".
2. Connect to a multiplayer server as a client.
3. Make any object invisible using the hotkey.
4. Other players still see the object.

#### Persistence: Server-side
1. Restart the server using a local autosave.
2. Reconnect as the client.
3. The object is visible again.

#### Persistence: Client-side
1. Save the map as the client.
2. Load the map in single-player mode.
3. The object is invisible as expected.

### PR DESCRIPTION
**Pull request by fidwell**
Refactored the Shortcuts class to call a method on TileInspector window class, instead of directly editing tile elements in the Shortcut class. This means that the invisibility shortcut will now use a game action, so it will work in multiplayer and for replays.

This implementation is based very closely on how the footpath window seems to handle shortcuts. I know ZehMatt mentioned in the issue that this system needs an overhaul, but at least this solution follows the current pattern.

Fixes #16453.

### CODE REVIEW

**Comment by deurklink** [comment]
I made a replay using this build and things that I made invisible and visible using the hotkeys now properly show that way in the replay as well. 
[replay.zip](https://github.com/OpenRCT2/OpenRCT2/files/12807146/replay.zip)

File: src/openrct2-ui/windows/TileInspector.cpp [hunk 1]
```diff
@@ -2369,3 +2377,10 @@ void WindowTileInspectorClearClipboard()
     if (window != nullptr)
         static_cast<TileInspector*>(window)->ClearClipboard();
 }
+
+void WindowTileInspectorKeyboardShortcutToggleInvisibility()
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/TileInspector.cpp:2381]: You are duplicating code here - you’re using WindowFindByClass() both here and in `ShortcutToggleVisibility()`.

I suggest moving this code to `ShortcutToggleVisibility()` and getting rid of `WindowTileInspectorKeyboardShortcutToggleInvisibility()`. (Assuming a team member agrees.)

Comment (fidwell) [reply src/openrct2-ui/windows/TileInspector.cpp:2381]: You're definitely right that there's redundant code. But I am not sure how to remove the reliance on `WindowTileInspectorKeyboardShortcutToggleInvisibility`. I can, however, remove `ShortcutToggleVisibility` entirely and have this function instead work like the ride construction shortcuts.

**Review by duncanspumpkin — APPROVED** [review]

File: src/openrct2-ui/windows/TileInspector.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2-ui/windows/TileInspector.cpp:2381]: That works too :+1:

**Pull request merged**